src/main/java/com/zaxxer/hikari/pool/HikariPool.java

```diff
@@ -400,22 +399,27 @@ final void releaseConnection(final PoolEntry poolEntry)
    /**
     * Permanently close the real (underlying) connection (eat any exception).
     *
-    * @param poolEntry the connection to actually close
+    * @param poolEntry poolEntry having the connection to close
+    * @param closureReason reason to close
+    * @param callClose false when the connection is dead because of un-recoverable exception and calling any method would result in exception, true otherwise
     */
-   final void closeConnection(final PoolEntry poolEntry, final String closureReason)
+   final void closeConnection(final PoolEntry poolEntry, final String closureReason, final boolean callClose)
```

Comment:
Regardless of whether the exception is un-recoverable or not, it is essential that we *always* call the driver's ``Connection.close()`` statement.  If we do not do this, we risk leaking system handles (Socket handles at a minimum), and we prevent the driver from cleaning up internal state such as tracked connections.

Additionally, even if the server disconnects or is unreachable, there is no guarantee that the ``close()`` itself would fail, that is entirely driver dependent.